lint-tasks.yml

```
@@ -1,127 +1,134 @@
 version: "3"
 
 vars:
+  G_LINT_CLANG_TIDY_DIR: "{{.G_BUILD_DIR}}/lint-clang-tidy"
+  G_LINT_VENV_DIR: "{{.G_BUILD_DIR}}/lint-venv"
   G_SRC_DIR: "{{.ROOT_DIR}}/src"
   G_SRC_CLP_FFI_JS_DIR: "{{.G_SRC_DIR}}/clp_ffi_js"
-  G_LINT_VENV_DIR: "{{.G_BUILD_DIR}}/lint-venv"
 
 tasks:
   check:
     cmds:
-      - task: "cpp-check"
-      - task: "yml-check"
+      - task: "check-cpp-full"
+      - task: "check-yaml"
+
+  check-cpp-diff:
+    cmds:
+      - task: "check-cpp-format"
+      - task: "check-cpp-static-diff"
+
+  check-cpp-full:
+    cmds:
+      - task: "check-cpp-format"
+      - task: "check-cpp-static-full"
+
+  check-no-cpp:
+    cmds:
+      - task: "check-yaml"
 
   fix:
     cmds:
-      - task: "cpp-fix"
-      - task: "yml-fix"
+      - task: "fix-cpp-full"
+      - task: "fix-yaml"
 
-  cpp-configs:
-    cmd: "{{.ROOT_DIR}}/tools/yscope-dev-utils/lint-configs/symlink-cpp-lint-configs.sh"
+  fix-cpp-diff:
+    cmds:
+      - task: "fix-cpp-format"
+      - task: "fix-cpp-static-diff"
 
-  cpp-check:
+  fix-cpp-full:
     cmds:
-      - task: "cpp-format-check"
-      - task: "cpp-static-check"
+      - task: "fix-cpp-format"
+      - task: "fix-cpp-static-full"
 
-  cpp-fix:
+  fix-no-cpp:
     cmds:
-      - task: "cpp-format-fix"
-      - task: "cpp-static-fix"
+      - task: "fix-yaml"
 
-  cpp-format-check:
+  check-cpp-format:
     sources: &cpp_format_source_files
       - "{{.G_SRC_CLP_FFI_JS_DIR}}/.clang-format"
-      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.cpp"
-      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.h"
-      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.hpp"
+      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*"
       - "{{.TASKFILE}}"
-      - "tools/yscope-dev-utils/lint-configs/.clang-format"
-    deps: ["cpp-configs", "venv"]
+      - "tools/yscope-dev-utils/exports/lint-configs/.clang-format"
+    deps: ["cpp-lint-configs", "venv"]
     cmds:
-      - task: "clang-format"
+      - task: ":utils:cpp-lint:clang-format"
         vars:
-          FLAGS: "--dry-run"
+          FLAGS: ["--dry-run"]
+          INCLUDE_FILENAME_PATTERNS: ["*.cpp", "*.h", "*.hpp", "*.inc"]
+          ROOT_PATHS: *cpp_format_source_files
```

Optional: technically this will make `find` try to (recursively) search from each individual file matching the `sources` list. I don't really see any correctness problem with it, but it is a bit weird logically.
The same is true in `fix-cpp-format`.

```suggestion
          ROOT_PATHS:
            - "{{.G_SRC_CLP_FFI_JS_DIR}}"
```